core/src/main/java/org/apache/druid/math/expr/vector/UnivariateFunctionVectorValueProcessor.java

```diff
@@ -74,5 +76,7 @@ public final ExprEvalVector<TOutput> evalVector(Expr.VectorInputBinding bindings
 
   abstract void processIndex(TInput input, int i);
 
+  abstract void processNull(int i);
```

Comment:
Instead of another abstract method call, I wonder how far we could get with a `.getValueForNull()` method that return a value that we can set?  With that setup, we would be able to call that once and then reuse the value, allowing for inlining and stuff.  It should work well for the scope of changes in this PR...

Reply:
it would have to return an object since we don't know if the value is `double` or `long`. Maybe the better thing to do is abstract out `long` and `double` versions of value processors and just have the same-ish code between them, though that's a bit more invasive of a change so i'm hesitant to make it right now. Would you be ok if i look into this as a follow-up PR?

Reply:
actually its not that bad, will go ahead and split them up now, it makes some other things nicer